scanner/src/lib/dataproducts/Battery_ManufacturingDataSheet_v0.1.svelte

```diff
@@ -0,0 +1,251 @@
+  <div class="subtitle">The details of the location of the battery manufacturing plant</div>
+  <DataRow label="City" value={data.manufacturingLocation.city} />
+  <DataRow label="Country" value={countryListAlpha3[data.manufacturingLocation.country]} />
+  <div class="divider" />
+  <div class="title no-bottom-margin">Round trip efficiency</div>
+  <div class="subtitle">The details of the round trip energy efficiency in energy storages</div>
+  <DataRow
+    label="Initial energy efficiency"
+    value={formatNumber(data.roundTripEfficiency.initialEnergyEfficieny, "%")}
+  />
+  <DataRow
+    label="Degraded energy efficiency"
+    value={formatNumber(data.roundTripEfficiency.degradedEnergyEfficiency, "%")}
+  />
+  <div class="divider" />
+  <div class="title no-bottom-margin">Voltage levels</div>
+  <div class="subtitle">The details of the voltage levels of the battery</div>
+  <DataRow label="Nominal" value={`${formatNumber(data.voltageLevels.nominalVoltage, "V")}`} />
+  <DataRow label="Maximum" value={`${formatNumber(data.voltageLevels.maximumVoltage, "V")}`} />
+  <DataRow label="Minimum" value={`${formatNumber(data.voltageLevels.minimumVoltage, "V")}`} />
+  <div class="divider" />
+  <div class="title no-bottom-margin">Temperature range</div>
+  <div class="subtitle">The details of the acceptable temperature values of the battery</div>
+  <DataRow
+    label="Temperature range"
+    value={`${formatNumber(data.temperatureRange.minimumTemperature, "°C")} to ${formatNumber(
+      data.temperatureRange.maximumTemperature,
+      "°C"
+    )}`}
+  />
+  <div class="divider" />
+  <div class="title no-bottom-margin">Expected lifetime</div>
+  <div class="subtitle">The details of the battery lifetime</div>
+  <DataRow label="Cycle life" value={formatNumber(data.expectedLifetime.cycleLife)} />
+  <DataRow label="Reference test" value={data.expectedLifetime.referenceTest} />
+  <DataRow label="Cycle rate" value={data.expectedLifetime.cycleRate} />
+  <div class="divider" />
+  <div class="title no-bottom-margin">Material composition</div>
+  <div class="subtitle">The details of the material composition of the battery</div>
+  <DataRow label="Chemistry" value={data.materialComposition.chemistry.join(", ")} />
+  <DataRow
+    label="Hazardous substances"
+    value={data.materialComposition.hazardousSubstances.join(", ")}
+  />
+  <DataRow
+    label="Critical raw materials"
+    value={data.materialComposition.criticalRawMaterials.join(", ")}
+  />
+  <div class="divider" />
+  <div class="title no-bottom-margin">Recycled content</div>
+  <div class="subtitle">The recycled content information present in the battery</div>
+  {#if data.recycledContent.length > 0}
+    <ul>
+      {#each data.recycledContent as content}
+        {#if content.substanceName.trim() && content.recyclingRate}
+          <li class="list-item">
+            {content.substanceName}: {formatNumber(content.recyclingRate, "%")}
+          </li>
+        {/if}
```

Comment:
also it's of course possible that due to rounding errors `recyclingRate` ends up being `0`

Reply:
Maybe we can check for `substanceName` alone cause it wouldn't make sense to have something like `-:  10%`

Reply:
Yes, it wouldn't make sense, but if for some reason the data being sent to us is `{substanceName: "", recyclingRate: 10}` then that is what we should show, instead of an empty list. It should be dealt on the definition level to ensure that `""` is not an allowed value.